tools/querytee/proxy_test.go

```diff
@@ -68,11 +68,116 @@ func testRequestTransformer2(r *http.Request, body []byte, _ *spanlogger.SpanLog
+		},
+		"multiple endpoints, preferred backend set and exists": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				PreferredBackend:                   "blah",
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: "",
+		},
+		"multiple endpoints, preferred backend set and does not exist": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				PreferredBackend:                   "blah-2",
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: "the preferred backend (hostname) has not been found among the list of configured backends",
+		},
+		"invalid endpoint": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "://blah",
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: `invalid backend endpoint ://blah: parse "://blah": missing protocol scheme`,
+		},
+		"multiple endpoints, secondary request proportion less than 0": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				PreferredBackend:                   "blah",
+				SecondaryBackendsRequestProportion: -0.1,
+			},
+			expectedError: "secondary request proportion must be between 0 and 1 (inclusive)",
+		},
+		"multiple endpoints, secondary request proportion greater than 1": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				PreferredBackend:                   "blah",
+				SecondaryBackendsRequestProportion: 1.1,
+			},
+			expectedError: "secondary request proportion must be between 0 and 1 (inclusive)",
+		},
+		"multiple endpoints, secondary request proportion 1 and preferred backend set": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				PreferredBackend:                   "blah",
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: "",
+		},
+		"multiple endpoints, secondary request proportion 1 and preferred backend not set": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				SecondaryBackendsRequestProportion: 1.0,
+			},
+			expectedError: "",
+		},
+		"multiple endpoints, secondary request proportion not 1 and preferred backend set": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				PreferredBackend:                   "blah",
+				SecondaryBackendsRequestProportion: 0.7,
+			},
+			expectedError: "",
+		},
+		"multiple endpoints, secondary request proportion not 1 and preferred backend not set": {
+			cfg: ProxyConfig{
+				BackendEndpoints:                   "http://blah,http://other-blah",
+				SecondaryBackendsRequestProportion: 0.7,
+			},
+			expectedError: "preferred backend must be set when secondary backends request proportion is not 1",
+		},
```

Comment:
Lets test for an endpoint like `https://blah.tld/promql/`. What would PreferredBackend `blah` give. Should it be `blah.tld` or include the path etc.

Reply:
Resolved in 4cbcc3e